### DISCUSSION TITLE
Upgrading my homebridge install, are there any "gotcha's?"

### DISCUSSION THREAD

**Discussion by ctgreybeard** [kuestess/homebridge-platform-insteonlocal#243] [General]
I have a working Homebridge installation (Pi 4, 2413 USB) but it hasn't been updated in a couple of years. Insteonlocal, for instance, is at 0.4.6 version, Homebridge is 0.4.50.

My plan is to do a fresh install in a separate directory of the code, copy over 
:
the config.json, 
persist and accessories subdirs, 
a couple of *persist.json files for weather-plus,
and what else?,
and then test.

What else should I copy to the new install?

What could go wrong? I want to avoid messing up my system. I don't have a lot of devices but they are the main lights in my house. In particular I would rather not lose my scenes and automations content.

Any advice would be appreciated!

**Comment by kuestess** [comment]
@ctgreybeard Welcome back!  You might try backing up your homebridge directory, and then updating in place.  You can always revert to a prior version.

**Comment by ctgreybeard** [comment]
When I first started using HB it was maintained using npm. My installation is currently running in a separate uid in a subdirectory of that ID.

Looking at the current install instructions HB is maintained via apt now and includes a spiffy new web UI. 

I'm not confident that my current installation will upgrade, using npm, to something that resembles the current capabilities.

But, it also looks like my original plan will not work either. I think I need to bite that bullet and just do a fully clean install of HB and start new. My Insteon config will transfer but I will probably lose the scene and room associations for my lights and switches. I can deal with that.

I also have a couple of other plugins but they can be easily recreated, I will still have the original config.json to refer to.

**Comment by kuestess** [comment]
@ctgreybeard You should still be able to update homebridge and the plugin using npm, and the web ui isn't required.  You can install the web ui separately if you want (its a plugin as well), see [https://github.com/oznu/homebridge-config-ui-x](https://github.com/oznu/homebridge-config-ui-x)

**Comment by ctgreybeard** [comment]
Yeah, but I would end up where I am now with a non-standard installation.

I am going to do a "standard" install and just put my cameras plugin into it. That will give me something to play with then I can decide if I want to move the Insteon plugin over or trash the new one and upgrade the old one.
